File: front/src/services/forum/ForumService.ts
import type {
  CreateForumPostInput,
  CreateForumReplyInput,
  CreateForumTopicInput,
  ForumPost,
  ForumReply,
  ForumTopic,
  UpdateForumPostInput,
  UpdateForumReplyInput,
} from './types.ts'

export interface ForumService {
  forumTopics(): Promise<ForumTopic[]>
  forumPosts(topicId?: string, limit?: number, offset?: number): Promise<ForumPost[]>
  forumPost(id: string): Promise<ForumPost | null>
  createForumPost(input: CreateForumPostInput): Promise<ForumPost>
  createForumReply(input: CreateForumReplyInput): Promise<ForumReply>
  updateForumPost(id: string, input: UpdateForumPostInput): Promise<ForumPost>
  updateForumReply(id: string, input: UpdateForumReplyInput): Promise<ForumReply>
  deleteForumPost(id: string): Promise<void>
  deleteForumReply(id: string): Promise<void>
  thankForumPost(postId: string): Promise<ForumPost>
  thankForumReply(replyId: string): Promise<ForumReply>
  createForumTopic(input: CreateForumTopicInput): Promise<ForumTopic>
  setForumTopicActive(topicId: string, active: boolean): Promise<ForumTopic>
  deleteForumTopic(topicId: string): Promise<void>
}

File: back/src/main/kotlin/com/jvmvstv_v/back/forum/service/ForumService.kt
package com.jvmvstv_v.back.forum.service

import com.jvmvstv_v.back.forum.model.CreateForumPostInput
import com.jvmvstv_v.back.forum.model.CreateForumReplyInput
import com.jvmvstv_v.back.forum.model.CreateForumTopicInput
import com.jvmvstv_v.back.forum.model.ForumPost
import com.jvmvstv_v.back.forum.model.ForumReply
import com.jvmvstv_v.back.forum.model.ForumTopic
import com.jvmvstv_v.back.forum.model.UpdateForumPostInput
import java.util.UUID

interface ForumService {
    fun topics(): List<ForumTopic>
    fun createTopic(input: CreateForumTopicInput): ForumTopic
    fun posts(topicId: Int?, limit: Int?, offset: Int?): List<ForumPost>
    fun post(id: UUID): ForumPost?
    fun createPost(input: CreateForumPostInput): ForumPost
    fun createReply(input: CreateForumReplyInput): ForumReply
    fun updatePost(id: UUID, input: UpdateForumPostInput): ForumPost
    fun updateReply(id: UUID, body: String): ForumReply
    fun thankPost(postId: UUID): ForumPost
    fun thankReply(replyId: UUID): ForumReply
    fun setTopicActive(topicId: Int, active: Boolean): ForumTopic
    fun deleteTopic(topicId: Int)
}
